cpp/src/arrow/ipc/message.cc

```diff
@@ -233,6 +235,16 @@ Status ReadMessage(io::InputStream* file, std::unique_ptr<Message>* message) {
     return Status::Invalid(ss.str());
   }
 
+  // If requested, align the file before reading the message.
+  if (aligned) {
+    int64_t offset;
+    RETURN_NOT_OK(file->Tell(&offset));
+    int64_t aligned_offset = PaddedLength(offset);
+    int64_t num_extra_bytes = aligned_offset - offset;
+    std::shared_ptr<Buffer> dummy_buffer;
+    RETURN_NOT_OK(file->Read(num_extra_bytes, &dummy_buffer));
```

Comment:
I'm not sure this is the right change. Shouldn't the alignment come from padding after the metadata?

Reply:
I'm not sure I understand the comment. This `Read` is used to skip over the padding. The metadata is read before this block, and the main contents are read after this block.